packages/drivers/odsp-driver/src/createFile.ts

```diff
@@ -97,6 +101,14 @@ export async function createNewFluidFile(
 
     const odspUrl = createOdspUrl({... newFileInfo, itemId, dataStorePath: "/"});
     const resolver = new OdspDriverUrlResolver();
+
+    // caching the converted summary
+    const snapshot: IOdspSnapshot = convertSummaryTreeToIOdspSnapshot(createNewSummary);
+    const value: ISnapshotCacheValue = { snapshot, sequenceNumber: 0 };
+    const odspResolvedUrl = await resolver.resolve({ url: odspUrl });
+    // eslint-disable-next-line @typescript-eslint/no-floating-promises
+    epochTracker.put(createCacheSnapshotKey(odspResolvedUrl), value);
+
     return resolver.resolve({ url: odspUrl });
```

Comment:
No need to resolve the url again. You can return the the above resolved url.

Reply:
ok. the url is already resolved, how can I return the promise?